src/dsp_tools/commands/xmlupload/resource_create_client.py

```diff
@@ -133,10 +135,22 @@ def make_values(p: XMLProperty) -> list[dict[str, Any]]:
         # To frame the json-ld correctly, we need one property used in the graph. It does not matter which.
         last_prop_name = None
 
+        str_value_func_mapper = {
+            "uri": SerialiseURI,
+            "color": SerialiseColor,
+        }
+
         for prop in resource.properties:
             match prop.valtype:
-                case "uri":
-                    properties_serialised.update(_serialise_uri_prop(prop, self.permissions_lookup))
+                # serialised as dict
+                case "uri" | "color" as val_type:
+                    transformed_prop = _transform_into_serialise_property_with_string_value(
```

Comment:
The name _transform_into_serialise_property_with_string_value() is a bit complicated. What about `_serialise_prop()`?

Reply:
You could use the nice name `_serialise_prop()` for now. And if later you need other functions for lists etc., you can name them accordingly, e.g. `_serialise_prop_special_case_1()`, `_serialise_prop_special_case_2()`, ...